### PR TITLE
Added compile commands generation for clangd lsp

### PR DESCRIPTION
**Pull request by mxple**

### CODE REVIEW

**Review by elsien-dev — APPROVED** [review]
Nice

File: CMakeLists.txt [hunk 1]
```diff
@@ -3,6 +3,8 @@ cmake_minimum_required(VERSION 3.14)
 project(lei3d)
 
 set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON CACHE INTERNAL "")
+set(CMAKE_BUILD_TYPE Debug)
```
Comment (kevinsadi) [inline CMakeLists.txt:7]: Good for us to specify to the build type. We'll want to make a note of this change when we start trying to get release executables.

File: CMakeLists.txt [hunk 2]
```diff
@@ -3,6 +3,8 @@ cmake_minimum_required(VERSION 3.14)
 project(lei3d)
 
 set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON CACHE INTERNAL "")
```
Comment (kevinsadi) [inline CMakeLists.txt:6]: lmao I was trying to figure out how to do this a while back, the clangd extension on vscode was still giving me problems despite doing this though? Is it working for you?

**Review by kevinsadi — APPROVED** [review]
Will need your wisdom on finding the compile_commands.json. I can't seem to generate it or if I am, I can't find the file. Game still works though, so LGTM.

**Pull request merged**